vocode/turn_based/synthesizer/coqui_synthesizer.py

```diff
@@ -1,29 +1,135 @@
-            "name": "unnamed",
-        }
+        # Split the text into chunks of less than MAX_TEXT_LENGTH characters
+        text_chunks = self.split_text(text)
+        # Synthesize each chunk and concatenate the results
+        audio_chunks = [self.synthesize_chunk(chunk) for chunk in text_chunks]
+        return sum(audio_chunks)  # type: ignore
+
+    def synthesize_chunk(self, text: str) -> AudioSegment:
+        url, headers, body = self.get_request(text)
+
+        # Get the sample
         response = requests.post(url, headers=headers, json=body)
         assert response.ok, response.text
         sample = response.json()
         response = requests.get(sample["audio_url"])
         return AudioSegment.from_wav(io.BytesIO(response.content))  # type: ignore
+
+    def split_text(self, text: str) -> List[str]:
+        sentence_enders = re.compile("[.!?]")
+        # Split the text into sentences using the regular expression
+        sentences = sentence_enders.split(text)
+        chunks = []
+        current_chunk = ""
+        for sentence in sentences:
+            # Strip leading and trailing whitespace from the sentence
+            sentence = sentence.strip()
+            # If the sentence is empty, skip it
+            if not sentence:
+                continue
+            # Concatenate the current chunk and the sentence, and add a period to the end
+            proposed_chunk = current_chunk + sentence
+            if len(proposed_chunk) > 250:
+                chunks.append(current_chunk.strip())
+                current_chunk = sentence + "."
+            else:
+                current_chunk = proposed_chunk + "."
+        # If there is a current chunk at the end, add it to the list of chunks
+        if current_chunk:
+            chunks.append(current_chunk.strip())
+        return chunks
+
+    async def async_synthesize(self, text: str) -> AudioSegment:
+        # This method is similar to the synthesize method, but it uses async IO to synthesize each chunk in parallel
+
+        # Split the text into chunks of less than MAX_TEXT_LENGTH characters
+        text_chunks = self.split_text(text)
+
+        # Create a list of tasks for each chunk using asyncio.create_task()
+        tasks = [
+            asyncio.create_task(self.async_synthesize_chunk(chunk))
+            for chunk in text_chunks
+        ]
+
+        # Wait for all tasks to complete using asyncio.gather()
+        audio_chunks = await asyncio.gather(*tasks)
+
+        # Concatenate and return the results
+        return sum(audio_chunks)  # type: ignore
+
+    async def async_synthesize_chunk(self, text: str) -> AudioSegment:
+        url, headers, body = self.get_request(text)
+
+        # Create an aiohttp session and post the request asynchronously using await
+        async with aiohttp.ClientSession() as session:
+            async with session.post(url, headers=headers, json=body) as response:
+                assert response.status == 201, (
```

Comment:
ok for now, but this sort of assert that is "expected" (not something that should never happen) - should probably be some other error class

Reply:
added linear followup